### PR TITLE
Issue #13345: Enable examples tests for NestedIfDepthCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for NestedIfDepthCheck

**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the **NestedIfDepthCheck** in the project. The implementation focuses on ensuring that NestedIfDepth checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs/checks/coding/nestedifdepth.xml [hunk 1]
```diff
@@ -44,58 +44,84 @@
...
         if (true) { // violation, nested if-else depth is 2 (max allowed is 1)
+          if (true) {} // violation, nested if-else depth is 2 (max allowed is 1)
+          else {}
         }
+      }
     }
-}
-        </source>
-        <p id="Example3-config">
-          To configure the check to allow nesting depth 3:
```
[the remark this answers is not in the document]
Comment (AmitKumarDeoghoria) [reply src/xdocs/checks/coding/nestedifdepth.xml:64]: Done

**Review by romani — APPROVED** [review]
Ok to merge.

Thanks a lot

**Pull request merged**